### PR TITLE
Adding default value to enabled flag on DrawItem

### PR DESCRIPTION
**Pull request by antonmic**
Will fix https://github.com/o3de/o3de/issues/17018

### CODE REVIEW

**Review by galibzon — APPROVED** [review]

**Review by moudgils — APPROVED** [review]

**Review by AMZN-Gene — APPROVED** [review]

File: Gems/Atom/RHI/Code/Include/Atom/RHI/DrawItem.h [hunk 1]
```diff
@@ -134,7 +134,7 @@ namespace AZ::RHI
             {
                 bool m_enabled : 1;    // Whether the Draw Item should render
             };
-            uint8_t m_allFlags;
+            uint8_t m_allFlags = 1;
```
Comment (VickyAtAZ) [inline Gems/Atom/RHI/Code/Include/Atom/RHI/DrawItem.h:137]: I think it's better to add m_enabled = true in constructor if `bool m_enabled : 1 = 1;` is not supported.
The union was added to have the potential of more than one flags. 
The m_allFlags = 1; here you have is to intent to set m_enabled = true which is implicit. 
If people add a flag before m_enabled in the future which could easily break the intention.

File: Gems/Atom/RHI/Code/Include/Atom/RHI/DrawItem.h [hunk 2]
```diff
@@ -134,7 +134,7 @@ namespace AZ::RHI
             {
                 bool m_enabled : 1;    // Whether the Draw Item should render
```
Comment (AMZN-Gene) [inline Gems/Atom/RHI/Code/Include/Atom/RHI/DrawItem.h:135]: Suggest adding a comment about the union being created to support any number of user flags (as long as they fit within 8bit), but current use only uses a 1-bit bitfield to determine if the draw item is enabled or not.

**Review by AMZN-Gene — APPROVED** [review]

**Review by VickyAtAZ — CHANGES_REQUESTED** [review]
Please check my comment.

File: Gems/Atom/RHI/Code/Include/Atom/RHI/DrawItem.h (hunk 1 above)
Comment (VickyAtAZ) [reply Gems/Atom/RHI/Code/Include/Atom/RHI/DrawItem.h:137]: Another option would be use m_disabled and set m_allFlags = 0; by default

**Review by VickyAtAZ — APPROVED** [review]

**Pull request merged**